common/amount.go

```diff
@@ -0,0 +1,79 @@
+package common
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Amount can be used to convert from fixed point to integer.
+// Amount assumes the smallest divisible unit is 1e-8
+type Amount int64
+
+func (a Amount) AsInt64() int64 {
+	return int64(a)
+}
+
+func (a Amount) MarshalJSON() ([]byte, error) {
+	return []byte(AmountToString(int64(a))), nil
+}
+
+func (a *Amount) UnmarshalJSON(data []byte) error {
+	v, err := StringToAmount(string(data))
+	if err != nil {
+		return err
+	}
+	*a = Amount(v)
+	return nil
+}
+
+func FloatToAmount(f float64) int64 {
+	return int64(f * 1e8)
+}
+
+// StringToAmount takes a number in decimal form, and converts it to
+// an integer * 1e8. Does not accept negative values
+func StringToAmount(amt string) (int64, error) {
```

Comment:
if you don't accept negative values why do you use int64 instead of uint64?

Reply:
The amounts in the code were all int64. I was just being consistent so I didn't need to do castings if I decided to use this.